src/vm/hints/sha256_utils.rs

```diff
@@ -0,0 +1,266 @@
+        .map_err(VirtualMachineError::MemoryError)?;
+    Ok(())
+}
+
+pub fn sha256_finalize(
+    vm: &mut VirtualMachine,
+    ids: &HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    let message: Vec<u8> = vec![0; 64];
+
+    let mut iv: [u32; SHA256_STATE_SIZE_FELTS] = [
+        0x6A09E667, 0xBB67AE85, 0x3C6EF372, 0xA54FF53A, 0x510E527F, 0x9B05688C, 0x1F83D9AB,
+        0x5BE0CD19,
+    ];
+
+    let iv_static: Vec<BigInt> = iv.iter().map(|n| bigint!(*n)).collect();
+
+    let new_message = GenericArray::clone_from_slice(&message);
+    compress256(&mut iv, &[new_message]);
+
+    let mut output: Vec<BigInt> = Vec::with_capacity(SHA256_STATE_SIZE_FELTS);
+
+    for new_state in iv {
+        output.push(bigint!(new_state));
+    }
+
+    let sha256_ptr_end = get_ptr_from_var_name(
+        "sha256_ptr_end",
+        ids,
+        &vm.memory,
+        &vm.references,
+        &vm.run_context,
+        hint_ap_tracking,
+    )?;
+
+    let mut padding: Vec<BigInt> = Vec::new();
+
+    for _ in 0..BLOCK_SIZE - 1 {
+        padding.extend(vec![BigInt::zero(); 16]);
+        padding.extend(iv_static.clone());
+        padding.extend(output.clone());
+    }
+
+    vm.segments
+        .write_arg(&mut vm.memory, &sha256_ptr_end, &padding, Some(&vm.prime))
+        .map_err(VirtualMachineError::MemoryError)?;
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::types::relocatable::MaybeRelocatable;
+    use crate::utils::test_utils::*;
+    use crate::vm::errors::memory_errors::MemoryError;
+    use crate::vm::hints::execute_hint::BuiltinHintExecutor;
+    use crate::vm::hints::execute_hint::HintReference;
+    use crate::vm::runners::builtin_runner::RangeCheckBuiltinRunner;
+    use crate::vm::vm_memory::memory::Memory;
+    use num_bigint::Sign;
+    static HINT_EXECUTOR: BuiltinHintExecutor = BuiltinHintExecutor {};
+
+    #[test]
+    fn sha256_input_one() {
+        let mut vm = vm_with_range_check!();
+        vm.memory = memory![((0, 1), 7)];
+        vm.run_context.fp = MaybeRelocatable::from((0, 2));
+        let ids = ids!["full_word", "n_bytes"];
+        vm.references = references!(2);
+
+        assert_eq!(
+            sha256_input(&mut vm, &ids, Some(&ApTracking::new())),
+            Ok(())
+        );
+
+        assert_eq!(
+            vm.memory.get(&MaybeRelocatable::from((0, 0))),
+            Ok(Some(&MaybeRelocatable::from(BigInt::one())))
+        );
+    }
+
+    #[test]
+    fn sha256_input_zero() {
+        let mut vm = vm_with_range_check!();
+        vm.memory = memory![((0, 1), 3)];
+        vm.run_context.fp = MaybeRelocatable::from((0, 2));
+        let ids = ids!["full_word", "n_bytes"];
+        vm.references = references!(2);
+
+        assert_eq!(
+            sha256_input(&mut vm, &ids, Some(&ApTracking::new())),
+            Ok(())
+        );
+
+        assert_eq!(
```

Comment:
This and the other memory assertions will need to be adjusted when the check_memory macro PR is merged, so that we start using them

Reply:
Resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/357/commits/6628578c9ca647bfdc85bf5b4494e2e130911833